### PR TITLE
[LLK Test Infra] Migrate remaining SFPU unary goldens to `_torch_unary` helper

### PR DESCRIPTION
**Pull request by jmacanovicTT**
### PR Category
Bug fix

### Summary
Relates to #45530

Continues the SFPU unary golden migration started for `log`, `sqrt`, `rsqrt`, `reciprocal`, `atanh`, `acosh`. 
Migrates the remaining ops:
- `exp`
- `exp2`
- `square`
- `cos`
- `sin`
- `tanh`
- `celu`
- `silu`
- `elu`
- `gelu`
- `asinh`
- `hardsigmoid`
- `sigmoid`
- `threshold`
- `relu_max`
- `relu_min`

to use `_torch_unary` helper. Switches internal compute precision to `fp32`, and makes the format-aware `inf→NaN` rule opt-in via an `overflow_to_nan` flag so the golden matches HW behavior on both classes of kernels (ones that emit `NaN` on `Float16` overflow vs. ones that pass `±inf` through).

Op-specific parameters (`alpha`, `threshold`, `replacement_value`) are additionally passed through `UnarySFPUGolden.__call__` instead of hardcoding them in method defaults. Removed `handle_infinite_numbers` helper since it's not used anymore.

### Notes for reviewers
- `overflow_to_nan` can be enabled per op and defaults to `True`. Use `True` for ops where HW returns `NaN` on `Float16` overflow (`exp`, `exp2`, `square`, `log`, `sqrt`, `rsqrt`, `atanh`, `acosh`). Use `False` for ops that pass `±inf` through unchanged (`celu`, `silu`, `elu`, `gelu`, `asinh`, `hardsigmoid`, `sigmoid`, `threshold`, `relu_max`, `relu_min`). 
- The dispatch loop was rewritten from a nested ternary into `if`/`elif`/`else` once it grew to four parameterized cases. Same behavior, easier to read.
- `_abs`, `_relu`, `_neg`, and `_fill` were intentionally not migrated. Their Python behavior is already simple and correct for IEEE special values, and `_fill` does not read its input, so it does not really match the unary-transform pattern.

FYI: @pmilenkovicTT @fvranicTT

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:jmacanovic/sfpu-golden-torch-migration)

### CODE REVIEW

File: tt_metal/tt-llk/tests/python_tests/helpers/golden_generators.py [hunk 1]
```diff
@@ -1914,17 +1917,21 @@ def __call__(
...
-                self.ops[operation](x, fill_const_value)
-                if operation == MathOperation.Fill
-                else self.ops[operation](x)
-            )
-            for x in result.tolist()[
-                ELEMENTS_PER_TILE * dest_idx : ELEMENTS_PER_TILE * dest_idx
-                + TILE_SIZE * iterations
-            ]
-        ]
+        op_res = []
```
Comment (fvranicTT) [inline tt_metal/tt-llk/tests/python_tests/helpers/golden_generators.py:1920]: With this proposal, the behavior is changed so that op is applied by looping over every scalar in the tile and building a fresh 0-d tensor per element with each helper doing `torch_fn(torch.tensor(x, dtype=torch.float32)).item()`. For a tile that's 1024 iterations including tensor allocations + dispatches + Python round-trips per call. The whole reason to move to torch primitives is that they're vectorized, so we're paying for the migration without getting the payoff.

File: tt_metal/tt-llk/tests/python_tests/helpers/golden_generators.py [hunk 2]
```diff
@@ -1850,6 +1850,9 @@ def __call__(
         iterations: int = None,
         dest_idx: int = 0,
         fill_const_value: float = 5,
+        alpha: float = 1.0,
+        threshold: float = 5.0,
+        replacement_value: float = 10.0,
```
Comment (fvranicTT) [inline tt_metal/tt-llk/tests/python_tests/helpers/golden_generators.py:1855]: Adding a parametrized op currently means touching three places: the self.ops dict, a new elif operation in (...) branch including the right args, and a new keyword here, in `__call__`. 

So an op's "shape" (which params it consumes) is spread across the dispatch loop and the call signature instead of living with the op. The duplicate `MathOperation.Tanh` entry (lines 1821 and 1826) is a small symptom of how easy it is for this dict to drift.

Let's see if we can optimize that. Give me some time to make a suggestion.
